src/Common/Scheduler/Nodes/FifoQueue.h

```diff
@@ -59,6 +60,9 @@ class FifoQueue final : public ISchedulerQueue
         std::lock_guard lock(mutex);
         if (is_not_usable)
             throw Exception(ErrorCodes::INVALID_SCHEDULER_NODE, "Scheduler queue is about to be destructed");
+
+        if (requests.size() >= static_cast<size_t>(info.queue_size))
+            throw Exception(ErrorCodes::SERVER_OVERLOADED, "Queue limit has been reached: {} of {}", requests.size(), info.queue_size);
```

Comment:
Let's give a hit on what exact limit has been reached:
```suggestion
            throw Exception(ErrorCodes::SERVER_OVERLOADED, "Workload limit `max_waiting_queries` has been reached: {} of {}", requests.size(), info.queue_size);
```